WordPress/Classes/Models/AbstractPost.swift

```diff
@@ -1,10 +1,81 @@
 import Foundation
 
 extension AbstractPost {
+
+    // MARK: - Status
+
+    @objc
+    var statusTitle: String? {
+        guard let status = self.status else {
+            return nil
+        }
+
+        return AbstractPost.title(for: status)
+    }
+
+    @objc
+    var remoteStatus: AbstractPostRemoteStatus {
+        get {
+            guard let remoteStatusNumber = remoteStatusNumber?.uintValue,
+                let status = AbstractPostRemoteStatus(rawValue: remoteStatusNumber) else {
+                    return .pushing
+            }
+
+            return status
+        }
+
+        set {
+            remoteStatusNumber = NSNumber(value: newValue.rawValue)
+        }
+    }
+
     class func title(for status: Status) -> String {
-        return AbstractPost.title(forStatus: status.rawValue)
+        return title(forStatus: status.rawValue)
+    }
+
+    /// Returns the localized title for the specified status.  Status should be
+    /// one of the `PostStatus...` constants.  If a matching title is not found
+    /// the status is returned.
+    ///
+    /// - parameter string: The post status value
+    /// - returns: The localized title for the specified status, or the status if a title was not found.
+    ///
+    @objc
+    class func title(forStatus status: String) -> String {
```

Comment:
Is there any specific reason why allowing these methods to be overridden is bad?  I'd rather make it more flexible unless there's a good reason not to.

Reply:
Here are some questions that went into my head:

#### When would we ever want to override `title(forStatus:)`? 

I guess we would need to override this if `Post` and `Page` would consider `PostStatusDraft` and the other enum values as different. The chances of that ever happening are slim since `Post` and `Page` are mostly interchangeable. If it does happen, what does that mean for the call sites like these?

https://github.com/wordpress-mobile/WordPress-iOS/blob/8c42fc3c29ac9e4bf06318fd67780f8d9929e0bf/WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m#L1068-L1070

https://github.com/wordpress-mobile/WordPress-iOS/blob/f29e1679b2985145bb8a0b4037a11ef32bfc93c6/WordPress/Classes/Models/AbstractPost.swift#L8-L14

Does that mean them we'll have to change them to do type-checking to ensure that they're calling the correct overidden routines, `[Post titleForStatus:]` or `[Page titleForStatus:]`? That is quite cumbersome and I'd wager we're probably going to change `titleForStatus` if that happens. There are better solutions than overriding this method. 

#### How expensive is it to change in the future? 

Keeping it strictly `static` makes the next developer to touch this code to think twice before overidding the method. They will have to think if it really makes sense, probably considering my points above. If they do end up overidding, the change is not expensive. 

It is more expensive to keep it open and have the next developer ask questions like _"It is open for overrides. Are the subclasses overidding it? Why?"_ This adds additional debugging time for every single time you pass by this code. Or worse, they could override it and forget about how the call sites are using this method. Human mistakes are expensive. 

#### Summary

In summary, I believe it's just simply better to have a closed API rather than opening everything. It's harder to roll back when we've opened something to the public and we later realized that it was wrong to do that. 

Happy to hear your thoughts. 🙂

Reply:
> When would we ever want to override title(forStatus:)?

For example if a subclass wants to return a different title, such as "Draft Page" for pages.

The truth is that ultimately `AbstractPost` should most likely not be where this method lives.  The title returned by this method is "specific" to one or several VCs, but definitely not universal.

But I digress.

> How expensive is it to change in the future?

Honestly it depends on what you want to do in the future.  If you wanted to override the title for a specific subtype of post it'd argue it'd be less expensive to have this be `class` methods.  And if the developer sees this is `static` they'd be asking themselves "why" - which as you mention above is just as expensive as asking the same question the other way around.

The bottom line is that while I appreciate the effort to make a point for a possible improvement... I don't think this *should* be `static` any more than it should be `class` methods, and again - I prefer to allow these methods to be overriden since there's a chance the title may not be the same for a page and a post.

Reply:
Actually, I'll just change this.

I still don't think this is strongly wrong, but I have no problem with this not being overridable for the time being.